services/workflows-service/src/transaction/test-utils/transaction-factory.ts

```diff
@@ -369,6 +375,14 @@ export class TransactionFactory {
     return factory;
   }
 
+  public date(dateFunction: (...args: any[]) => Date, ...args: any[]) {
+    const factory = this.clone();
+
+    factory._fakeDateByFn = () => dateFunction(...args);
+
+    return factory;
+  }
+
```

Comment:
@MatanYadaev @alonp99  FYI: 
The purpose of this change is to allow each generated transaction to have a unique date based on the provided `dateFunction`.

 * 
Previously, all transaction records might have been using the same date generate by the faker, leading to a lack of variation in transaction timestamps. all the timestamps were the same!
 * 
By introducing the `date` method with a `dateFunction` (such as `faker.date.past()`, `faker.date.recent()`, etc.), we ensure that each transaction can have its own random date, adding variability to the generated data.
 * 
The method works by cloning the factory instance and assigning the custom date logic through `_fakeDateByFn`, which is executed when a date is needed for each record.